drivers/bme280/bme280.c

```diff
@@ -0,0 +1,453 @@
+#define BME280_DIG_P9_MSB_REG                   0x9F
+
+#define BME280_DIG_H1_REG                       0xA1
+
+#define BME280_CHIP_ID_REG                      0xD0 /* Chip ID */
+#define BME280_RST_REG                          0xE0 /* Softreset Reg */
+
+#define BME280_DIG_H2_LSB_REG                   0xE1
+#define BME280_DIG_H2_MSB_REG                   0xE2
+#define BME280_DIG_H3_REG                       0xE3
+#define BME280_DIG_H4_MSB_REG                   0xE4 /* H4[11:4] */
+#define BME280_DIG_H4_H5_REG                    0xE5 /* H5[3:0]  H4[3:0] */
+#define BME280_DIG_H5_MSB_REG                   0xE6 /* H5[11:4] */
+#define BME280_DIG_H6_REG                       0xE7
+
+#define BME280_CTRL_HUMIDITY_REG                0xF2 /* Ctrl Humidity Reg */
+#define BME280_STAT_REG                         0xF3 /* Status Reg */
+#define BME280_CTRL_MEAS_REG                    0xF4 /* Ctrl Measure Reg */
+#define BME280_CONFIG_REG                       0xF5 /* Configuration Reg */
+#define BME280_PRESSURE_MSB_REG                 0xF7 /* Pressure MSB */
+#define BME280_PRESSURE_LSB_REG                 0xF8 /* Pressure LSB */
+#define BME280_PRESSURE_XLSB_REG                0xF9 /* Pressure XLSB */
+#define BME280_TEMPERATURE_MSB_REG              0xFA /* Temperature MSB */
+#define BME280_TEMPERATURE_LSB_REG              0xFB /* Temperature LSB */
+#define BME280_TEMPERATURE_XLSB_REG             0xFC /* Temperature XLSB */
+#define BME280_HUMIDITY_MSB_REG                 0xFD /* Humidity MSB */
+#define BME280_HUMIDITY_LSB_REG                 0xFE /* Humidity LSB */
+
+static int read_calibration_data(bme280_t* dev);
+static int do_measurement(bme280_t* dev);
+static uint8_t get_ctrl_meas(bme280_t* dev);
+static uint8_t get_status(bme280_t* dev);
+static uint8_t read_u8_reg(bme280_t* dev, uint8_t offset);
+static void write_u8_reg(bme280_t* dev, uint8_t offset, uint8_t b);
+static uint16_t get_uint16_le(const uint8_t *buffer, size_t offset);
+static int16_t get_int16_le(const uint8_t *buffer, size_t offset);
+
+static void dump_buffer(const char *txt, uint8_t *buffer, size_t size);
+
+/**
+ * @brief   Allocation of memory for device descriptors
+ */
+static bme280_t bme280_devs[BME280_NUMOF];
+
+/**
+ * @brief   Fine resolution temperature value, also needed for pressure and humidity.
+ */
+static int32_t t_fine;
+
+/**
+ * @brief   The measurement registers, including temperature, pressure and humidity
+ *
+ * A temporary buffer for the memory map 0xF7..0xFE
+ * These are read all at once and then used to compute the three sensor values.
+ */
+static uint8_t measurement_regs[8];
+
+/*---------------------------------------------------------------------------*
+ *                          BME280 Core API                                  *
+ *---------------------------------------------------------------------------*/
+
+int bme280_init(bme280_t* dev, i2c_t i2c, uint8_t addr, const bme280_setting_t* settings)
```

Comment:
Could you change this to 
```c
int bme280_init(bme280_t *dev, const bme280_params_t *params);
```
as done for example in #6153 or for the bh1750fvi, the dht, or the lpd8808 drivers?! This is the preferred way of device driver initialization currently in RIOT. I am sorry that there is no documentation on this (yet), planing to have a 'driver writing guide' in for the next release...